### DISCUSSION TITLE
Lesson 4: deploy.py : Plz step-dev I am stuck

### DISCUSSION THREAD

**Discussion by GauPerrin** [smartcontractkit/full-blockchain-solidity-course-py#1309] [Q&A]
Hey I am at that point when supposed to call "python deploy.py"
I've corrected the video issue 
"from solcx import compile_standard, install_solc
install_solc("0.6.0") "

Yet the terminal returns me this, any idea on how could I get the same result as on tne video ? 
Thx :)

Traceback (most recent call last):
  File "C:\Users\Utilisateur\demos\web3_py_simple_storage\deploy.py", line 10, in <module>
    compiled_sol = compile_standard(        
  File "C:\Users\Utilisateur\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\solcx\main.py", line 375, in compile_standard
    stdin=json.dumps(input_data),
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1008.0_x64__qbz5n2kfra8p0\lib\json\__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)     
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1008.0_x64__qbz5n2kfra8p0\lib\json\encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
oftwareFoundation.Python.3.10_3.10.1008.0_x64__qbz5n2kfra8p0\lib\json\encoder.py", line 179, in default    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type set is not JSON serializable

**Comment by cromewar** [comment]
Hello there, please share the code you are using to deploy and also try using a different python version on a virtualenv, thanks.

**Reply by GauPerrin** [reply]
```Python
import json

# In the video, we forget to `install_solc`
# from solcx import compile_standard
from solcx import compile_standard, install_solc

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

# We add these two lines that we forgot from the video!
print("Installing...")
install_solc("0.6.0")

# Solidity source code
compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {
                    "*": ["abi", "metadata", "evm.bytecode", "evm.bytecode.sourceMap"]
                }
            }
        },
    },
    solc_version="0.6.0",
)

with open("compiled_code.json", "w") as file:
    json.dump(compiled_sol, file)
    
    
```
   Here's the code, I am actually using VSC
   Thx :)

**Reply by ghost** [reply]
Thanks a lot man! I literaly trying to solve this for 3 hours, i'm new to this, I've installed wsl, did a lot of stuff and the answer was here, thanks man. I appreciate it.

**Reply by cromewar** [reply]
@Arcangel505 your code seems to be ok, this should be related to your python version, could you try using a virtualenv with python version 3.8.

**Reply by GauPerrin** [reply]
Ok i'll try it thanks, I'll tell you if I work it through :)

**Reply by GauPerrin** [reply]
I am on windows so I have Python 3.10.3

**Reply by GauPerrin** [reply]
@TecnoGames120  you managed to work this through ?

**Reply by ghost** [reply]
> @TecnoGames120 you managed to work this through ?

Yeah, I manage to do it, i dont khow how, but i did it. In my case i just needed to install the solidity compiler. 

from solcx import compile_standar, install_solc
install_solc("latest")

But your problem its different, how its going to you? You did it?

**Comment by n4n0b1t3** [comment]
When starting this lesson please first **clone** the file from the repository as described here: https://github.com/PatrickAlphaC/web3_py_simple_storage

You need to read the Readme, hence the name. https://github.com/PatrickAlphaC/web3_py_simple_storage#readme 

Try to run this first, and then start creating your own version, if your version doesn't run, compare it carefully with the provided and running version from the repo before asking a question. As long as the repo version runs on your computer and your file doesn't you will have a simple typo or something you can fix yourself.

**Reply by GauPerrin** [reply]
Yes sure thx, I had my mind away !

**Comment by Noveleader** [comment]
Hey @Arcangel505 try using full path I sometimes face same problem.
Like this:
```
with open(r"c:\Users\hp\VS code\Web_3\compiled_code.json", "w") as file:
        json.dump(compiled_sol, file)
```

**Reply by GauPerrin** [reply]
Thx actually "It tells me that it can't find the folder specified"
However it opened me a compiled_code.json folder I must admit as a newbie I am a bit confused haha

**Reply by Noveleader** [reply]
So it works or not finally?

**Reply by cromewar** [reply]
Yes @Arcangel505 Could you confirm if the problem is solved or no?
